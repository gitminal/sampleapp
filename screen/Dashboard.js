import { View, Text, StyleSheet, FlatList, ActivityIndicator } from 'react-native'
import React, { useEffect } from 'react'
import { useSelector, useDispatch } from 'react-redux'
import Colors from '../constant/Colors'
import { GET_APISHOW, getproduct } from '../redux/action';
import ProductComponent from '../Component/ProductComponent';
import Seperator from './Seperator';
export default function Dashboard({ navigation }) {
  const { product, error, loading } = useSelector(state => state.productReducer);
  console.log("product.....", product);
  const dispatch = useDispatch();
  const fetchResult = () => dispatch(getproduct());
  useEffect(() => {
    dispatch({
      type: GET_APISHOW,
    })
    fetchResult();
  }, []);
 

  const renderItem = ({ item }) => {
    function pressHandler() {
      navigation.navigate("ProductDetail",
        {
          item: item,
        })
    }

    console.log("item", item);
    return (
      <ProductComponent item={item} onPress={pressHandler} />
    )
  }
  return (
    <View style={styles.container}>
      {loading ?
        <ActivityIndicator style={styles.activity} />
        : error ? <Text> {error}</Text> :
          <FlatList
            data={product}
            keyExtractor={(item, index) => index.toString()}
            renderItem={renderItem}
            ItemSeparatorComponent={Seperator}
          />
      }
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingHorizontal: 10,
    backgroundColor: Colors.white
  },
  activity: {
    size: "large",
    color: Colors.gray,
    justifyContent: "center",
    flex:1
  },
 
  label: {
    fontSize: 14,
    paddingLeft: 10,
    alignSelf: "flex-start",
    paddingLeft: 15,
    fontWeight: "bold",
    marginTop: 10,

  },
  labelproduct: {
    fontSize: 15,
    paddingLeft: 15,
    fontWeight: "bold"

  },
 
})