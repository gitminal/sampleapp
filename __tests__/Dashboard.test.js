import {render,screen } from 'react-native-testing-library'
import React from 'react';
import enzyme from 'enzyme';
import { Provider } from 'react-redux'
import {mount} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk';
import Dashboard from '../screen/Dashboard';
import { FlatList ,View} from 'react-native';
describe('Dashboard ', () => {
  const initialState = {
    productReducer: {
       product: [],
       loading:false,
       error:'',
    },
  };
  const middlewares = [thunk];
  const mockStoreConfigure = configureMockStore(middlewares);
  const store = mockStoreConfigure({ ...initialState });
  const originalDispatch = store.dispatch;
  store.dispatch = jest.fn(originalDispatch)
    enzyme.configure({ adapter: new Adapter() });
  const mockDataproduct = [
    {
      id: 'id-1',
      name: 'Todo-1',
    },
    {
      id: 'id-2',
      name: 'Todo-2',
    },
    {
      id: 'id-3',
      name: 'Todo-3',
    },
  ]
  test('flatlist component should be render', () => {
    const createTestProps = (props) => ({
      navigation: {
        navigate: jest.fn(),
      },
      ...props
    });
    const componentTree = render(
      <Provider store={store}>
      <Dashboard  todos={mockDataproduct}  />
  </Provider>
    )
    expect(componentTree.UNSAFE_getAllByType(FlatList).length).toBe(1)
    expect(componentTree.UNSAFE_getAllByType( View).length).toMatchSnapshot();

  })

})
