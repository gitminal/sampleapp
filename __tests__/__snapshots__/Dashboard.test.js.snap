// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard  flatlist component should be render 1`] = `2`;
